tests/system/test_crud.py

```diff
@@ -1488,3 +1488,34 @@ class Container(ndb.Model):
 
     entity = entity.key.get()
     assert entity.child.foo == "bar"
+
+
+@pytest.mark.usefixtures("client_context")
+def test_local_structured_property_with_inheritance(dispose_of):
+    """Regression test for #523
+
+    https://github.com/googleapis/python-ndb/issues/523
+    """
+
+    class Base(ndb.Model):
+        pass
+
+    class SubKind(Base):
+        foo = ndb.StringProperty()
+
+    class Container(ndb.Model):
+        children = ndb.LocalStructuredProperty(Base, repeated=True)
+
+    entity = Container()
+
+    subkind = SubKind(foo="bar")
+    subkey = subkind.put()
+
+    entity.children.append(subkind)
+    key = entity.put()
```

Comment:
Why are we doing a `put` here? Since a `LocalStructuredProperty` is just stored on the parent entity, don't we effectively have two independent copies of this entity now?

Reply:
I was trying to duplicate the user problem as exactly as possible. It might be good to take that out, though.

Reply:
@chrisrossi removed extra put()